src/sage/rings/species.py

```diff
@@ -0,0 +1,2337 @@
+        r"""
+        Initialize an atomic species.
+
+        INPUT:
+
+        - ``dis`` -- a directly indecomposable permutation group
+        - ``domain_partition`` -- a `k`-tuple of ``frozenset``s,
+          where `k` is the arity, representing the assignment of each
+          element of the domain of ``dis`` to a sort
+
+        TESTS::
+
+            sage: from sage.rings.species import AtomicSpecies
+            sage: A = AtomicSpecies("X")
+            sage: G = PermutationGroup([[(1,3),(4,7)], [(2,5),(6,8)], [(1,4),(2,5),(3,7)]])
+            sage: TestSuite(A(G)).run()
+
+            sage: loads(dumps(A(G))) is A(G)
+            True
+        """
+        Element.__init__(self, parent)
+        self._dis = dis
+        self._dompart = domain_partition
+        self._mc = tuple([len(v) for v in self._dompart])
+        self._tc = sum(self._mc)
+
+    def _repr_(self):
+        r"""
+        Return a string representation of ``self``.
+
+        TESTS::
+
+            sage: from sage.rings.species import AtomicSpecies
+            sage: A = AtomicSpecies("X, Y")
+            sage: G = PermutationGroup([[(1,2),(3,4),(5,6),(7,8,9,10)]]); G
+            Permutation Group with generators [(1,2)(3,4)(5,6)(7,8,9,10)]
+            sage: a = A(G, {0: [1,2,3,4], 1: [5,6,7,8,9,10]}); a
+            {((1,2,3,4)(5,6)(7,8)(9,10),): ({5, 6, 7, 8}, {1, 2, 3, 4, 9, 10})}
+            sage: a = A(G, {1: [1,2,3,4,5,6,7,8,9,10]}); a
+            {((1,2,3,4)(5,6)(7,8)(9,10),): ({}, {1, 2, 3, 4, 5, 6, 7, 8, 9, 10})}
+        """
+        if self.parent()._arity == 1:
+            return "{" + f"{self._dis.gens()}" + "}"
+        dompart = ', '.join("{" + repr(sorted(b))[1:-1] + "}"
+                            for b in self._dompart)
+        return "{" + f"{self._dis.gens()}: ({dompart})" + "}"
+
+    def grade(self):
+        r"""
+        Return the grade of ``self``.
+
+        EXAMPLES::
+
+            sage: from sage.rings.species import AtomicSpecies
+            sage: A = AtomicSpecies("X, Y")
+            sage: G = PermutationGroup([[(1,2),(3,4),(5,6),(7,8,9,10)]])
+            sage: a = A(G, {0: [1,2,3,4], 1: [5,6,7,8,9,10]})
+            sage: a.grade()
+            [4, 6]
+        """
+        S = self.parent().grading_set()
+        return S(self._mc)
+
+    def __lt__(self, other):
+        r"""
+        Return whether ``self`` is less than ``other``.
+
+        ``self`` is less than or equal to ``other`` if it is
+        conjugate to a subgroup of ``other`` in the parent group.
+
+        EXAMPLES::
+
+            sage: from sage.rings.species import AtomicSpecies
+            sage: A = AtomicSpecies("X")
+            sage: A(DihedralGroup(4)) < A(CyclicPermutationGroup(4))
+            True
+
+        We create the poset of atomic species of degree four::
+
+            sage: P = Poset([A.subset(4), lambda b, c: b <= c])
+            sage: len(P.cover_relations())
+            7
+            sage: P.cover_relations()  # random
```

Comment:
This isn't really a useful example (and doctest) being random output. Is there some property you could test instead?